test/test_tensor.h

```diff
@@ -129,6 +129,62 @@ TEST(tensor, check_bounds) {
     EXPECT_THROW(tensor.host_ptr(1,0,0,1), nn_error);
 }
 
+TEST(tensor, view1) {
+
+    Tensor<float_t, 4> t1({2,2,2,2});
+    Tensor<float_t, 4> t2 = t1.view({2,4});
+    Tensor<float_t, 4> t3 = t1.view({0}, {2,4});
+
+    EXPECT_TRUE(!t1.isView());
+    EXPECT_TRUE( t2.isView());
+    EXPECT_TRUE( t3.isView());
+
+    EXPECT_EQ(t1.size(), size_t(2*2*2*2));
+    EXPECT_EQ(t2.size(), size_t(2*4));
+    EXPECT_EQ(t3.size(), size_t(2*4));
+}
+
+TEST(tensor, view2) {
+
+    Tensor<float_t, 4> t1({2,2,2,2});
+
+    EXPECT_NO_THROW(t1.view({2,4,0,0}));
+    EXPECT_NO_THROW(t1.view({0,0,0,0}, {2,4,0,0}));
+
+    // test that num parameters exceed to num of dimensions
+    EXPECT_THROW(t1.view({2,4,0,0,0}), nn_error);
+    EXPECT_THROW(t1.view({0,0,0,0},   {2,4,0,0,0}), nn_error);
+    EXPECT_THROW(t1.view({2,4,0,0,0}, {1,2,0,0}),   nn_error);
+    EXPECT_THROW(t1.view({2,4,0,0,0}, {1,2,0,0,0}), nn_error);
+}
+
+TEST(tensor, view3) {
+
+    Tensor<float_t, 4> t1({2,2,2,2});
+
+    // test that cannot generate view bigger than current tensor
+    EXPECT_THROW(t1.view({4,5}), nn_error);
+    EXPECT_THROW(t1.view({1}, {4,4}), nn_error);
+}
+
+TEST(tensor, view4) {
+
+    Tensor<float_t, 4> t1({2,2,2,2});
+    Tensor<float_t, 4> t2 = t1.view({2,4});
+
+    for (size_t i = 0; i < 8; ++i) {
+        EXPECT_EQ(t1.host_data()[i], float_t(0.0));
+        EXPECT_EQ(t2.host_data()[i], float_t(0.0));
+    }
+
+    t2.fill(float_t(2.0));
```

Comment:
@edgarriba I mean that if for some reason `host_data[i]` returns `i+2`nd element instead of `i`th, it will still pass tests. So I suppose to fill tensor with something like `0,1,2,3,4,5,6,7,8`... such that you make sure right element access.